modules/nw-operator-vsphere-multiple-nic.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_vsphere/ipi/installing-vsphere-installer-provisioned-network-customizations.adoc
+// * installing/installing_vsphere/upi/installing-vsphere-network-customizations.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-operator-vsphere-multiple-nic_{context}"]
+= Specifying multiple network interface controllers (NICs)
+
+Before you install an {product-title} cluster on a {vmw-short} host, you can specify multiple network interface controllers (NICs) for a networking implementation so that the {vmw-short} cloud controller manager (CCM) can select the appropriate NIC for a given networking situation. The {vmw-full} can use the NIC for managing pods and services on your cluster. For this configuration, you must specify internal and external Classless Inter-Domain Routing (CIDR) implementations in the {vmw-short} cloud controller manager (CCM) configuration. Each CIDR implementation lists an IP address range that the CCM uses to determine what NICs interact with traffic from internal and external networks.
+
+If you do not configure internal and external CIDR implementations in the {vmw-short} CCM configuration, the {vmw-short} CCM selects the first NIC that the host operating system reports as the primary network interface. This configuration can cause new nodes that associate with a `MachineSet` object with a single NIC to become unusable as each new node receives the `node.cloudprovider.kubernetes.io/uninitialized` taint. In certain situations, the reported network interface might not be the expected network interface. These situations can cause communication issues with the Kubernetes API server that can cause installation of the cluster to fail.
+
+[IMPORTANT]
+====
+If you configured multiple NICs for a network implementation but intend to use only a single NIC for your current needs, ensure that you specify the same IP address for the internal and external CIDR implementations. This configuration ensures that the CCM selects your preferred NIC and not the first NIC that the host operating system reports.
+====
+
+.Example error message because of the {vmw-short} CCM selecting an incompatible NIC
+[source, text]
+----
+ERROR Bootstrap failed to complete: timed out waiting for the condition
+ERROR Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane.
+----
+
+.Prerequisites
+
+* You created Kubernetes manifest files for your {product-title} cluster.
+
+.Procedure
+
+. Enter the following command in the {oc-first} tool to view detailed information, especially `Internal-IP` or `External-IP information, for each node that exists in your cluster. You will need this information in a the `nodeNetworking:` section of the `nodeNetworking` object.
++
+[source,terminal]
+----
+$ oc get nodes -o wide
```

Comment:
This isn't going to work for day 1 configuration. There won't be a cluster to look at.

Even if we're assuming that they got here because the first attempt failed and they have a broken partial cluster to look at, I'm not sure we can rely on the output from this command since the IP selection is by definition wrong if they need this procedure.  If the IPs were correctly assigned to internal and external then they wouldn't need to override the behavior.

I think we need to explain how to select values for the fields independent of any existing cluster. Even though there may be upgrade cases where a valid cluster config exists for reference, we can't count on it.

Reply:
OK. I did not tailor these docs as a DAY-2 activity. Should I?

Reply:
Thanks, @cybertron . Removed:

```
. Enter the following command in the {oc-first} tool to view detailed information, especially `Internal-IP` or `External-IP information, for each node that exists in your cluster. You will need this information in a the `nodeNetworking:` section of the `nodeNetworking` object.
+
[source,terminal]
----
$ oc get nodes -o wide
----
```

Reply:
"I think we need to explain how to select values for the fields independent of any existing cluster. Even though there may be upgrade cases where a valid cluster config exists for reference, we can't count on it."

^^ I think this is explained?